File: src/main/java/com/tairan/zt/hadoop/SimpleApp.java
package com.tairan.zt.hadoop;

/**
 * Created by hzzt on 2016/8/5.
 */
import org.apache.spark.api.java.*;
import org.apache.spark.SparkConf;
import org.apache.spark.api.java.function.Function;

public class SimpleApp {
	public static void main(String[] args) {
		String logFile = "hdfs://172.30.251.229:9000/user/hzzt/input/hello.txt"; // Should be some file on your system
		SparkConf conf = new SparkConf().setAppName("SimpleApp").setMaster("spark://172.30.251.229:7077");
		JavaSparkContext sc = new JavaSparkContext(conf);
		sc.addJar("D:\\workspace\\SparkTest\\out\\artifacts\\sparktest_jar\\sparktest.jar");
		JavaRDD<String> logData = sc.textFile(logFile).cache();

		long numAs = logData.filter(new Function<String, Boolean>() {
			public Boolean call(String s) { return s.contains("a"); }
		}).count();

		long numBs = logData.filter(new Function<String, Boolean>() {
			public Boolean call(String s) { return s.contains("b"); }
		}).count();

		System.out.println("Lines with a: " + numAs + ", lines with b: " + numBs);
	}
}

File: src/main/scala/com/tairan/zt/SimpleApp.scala
package com.tairan.zt

import org.apache.spark.{SparkContext, SparkConf}

/**
 * Created by hzzt on 2016/8/5.
 */
object SimpleApp {
  def main(args: Array[String]) {
    val conf = new SparkConf().setAppName("SimpleApp").setMaster("spark://172.30.251.229:7077")
    val sc = new SparkContext(conf)
    sc.addJar("D:\\workspace\\SparkTest\\out\\artifacts\\sparktest_jar\\sparktest.jar")
    val logData = sc.textFile("hdfs://172.30.251.229:9000/user/hzzt/input/hello.txt", 2).cache()
    val numAs = logData.filter(line => line.contains("a")).count()
    val numBs = logData.filter(_.contains("b")).count()
    println("Lines with a: %s, Lines with b: %s".format(numAs, numBs))
    sc.stop()




  }
}
